<?xml version="1.0"?>
<launch>

  <!-- visualize and edit teached grasps dumped into yaml file with rviz -->

  <arg name="object_name" default="relay"/>
  <arg name="grasps_yaml_path" default="$(find mobipick_pick_n_place)/config/grasplan/object_grasps"/>
  <arg name="obj_pkg_name" default="pbr_objects" />

  <!-- apply a transform to the object -->
  <arg name="object_transform_linear_x" default="0.0" />
  <arg name="object_transform_linear_y" default="0.0"/>
  <arg name="object_transform_linear_z" default="0.0"/>
  <arg name="object_transform_angular_roll" default="0.0"/>
  <arg name="object_transform_angular_pitch" default="0.0"/>
  <arg name="object_transform_angular_yaw" default="0.0"/>

  <!-- rqt_grasplan does not work if use_sim_time is set to true -->
  <param name="/use_sim_time" value="False" />

  <!-- rqt to edit grasps -->
  <node pkg="grasplan" type="rqt_grasplan" name="grasp_editor" output="screen" required="true" >
    <param name="grasps_yaml_path" value="$(arg grasps_yaml_path)" />
    <param name="object_name" value="$(arg object_name)" />
    <param name="transform_linear_x" value="$(arg object_transform_linear_x)" />
    <param name="transform_linear_y" value="$(arg object_transform_linear_y)" />
    <param name="transform_linear_z" value="$(arg object_transform_linear_z)" />
    <param name="transform_angular_roll" value="$(arg object_transform_angular_roll)" />
    <param name="transform_angular_pitch" value="$(arg object_transform_angular_pitch)" />
    <param name="transform_angular_yaw" value="$(arg object_transform_angular_yaw)" />
    <!-- the ros pkg where the object mesh can be found, {obj_pkg_name}/meshes/{object_name}.dae -->
    <param name="obj_pkg_name" value="$(arg obj_pkg_name)" />
  </node>

  <!-- visualize gripper in rviz as marker -->
  <arg name="gripper_transforms_config_file"
       default="$(find mobipick_pick_n_place)/config/gripper_open_transformations.yaml" />
  <arg name="pose_array_topic" default="/grasp_editor/grasp_poses" />
  <arg name="global_reference_frame" default="object"/>
  <include file="$(find grasplan)/launch/visualization/rviz_gripper_visualizer.launch">
    <arg name="gripper_transforms_config_file"
         value="$(arg gripper_transforms_config_file)"/>
    <arg name="pose_array_topic" value="$(arg pose_array_topic)"/>
    <arg name="global_reference_frame" value="$(arg global_reference_frame)"/>
  </include>

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find grasplan)/config/rviz/grasp_editor.rviz" required="true" />

</launch>
